subsys/bluetooth/services/ots/ots_internal.h

```diff
@@ -66,6 +70,12 @@ struct bt_gatt_ots_indicate {
 	bool is_enabled;
 };
 
+struct bt_ots_dir_list {
+	struct net_buf_simple net_buf;
+	struct bt_gatt_ots_object *dir_list_obj;
+	uint8_t _content[DIR_LIST_MAX_SIZE];
+};
```

Comment:
Please move this definition to ots_dir_list header and simply include it here (same for new defines).

Reply:
That was my original intend as well, but then we'd have to deal with circular dependencies as this needs the declaration of struct bt_gatt_ots_object.

Possible solutions, if we still want to move it, is to add an opaque type or to use void for the obj pointer type.

Reply:
Hmm, thanks for explaining.

We could have ots_object_internal.h header with Object structure definition and all its deps.

However, we would still have a circular dependency because the OTS instance is used in the ots_dir_list_internal.h. Is it possible to refactor the ots_dir_list_internal API so that it does not have to access the full OTS instance? I think you are using only three instance fields in the Directory listing source file: dir_list, cur_obj and obj_manager.

I also think that it is a good practice to pass only data that are required by the receiving module and nothing more. I know that access to the full OTS state is convenient but it could also be more error-prone. For example, you may by accident change the OACP state in the Directory listing related code.

Reply:
I am only using a few instance fields indeed, so it should be possible. 

On the other hand, if the function requires e.g. the `obj_manager`, the `dir_list` and `cur_obj`, there is likewise a risk that one of these parameters might be passed from the wrong OTS instance, and we would have to (or should) add checks/asserts that the different values are from the same OTS instance, which would basically do a reverse lookup (using e.g. `CONTAINER_OF`) to get the OTS instance. (in reality we could do that as a solution as well, and just take the dir_list object as arguments, but that's also kinda weird in my opinion).

Reply:
Yes, sure, I can see your point. I still think that it would be useful to change the code according to my suggestion.

Any opinions on this from the other reviewers? I think that outside feedback would help to resolve this discussion faster.

Reply:
@kapi-no What do you think? Should I refactor to only require the needed parameters, or keep as is?

Reply:
@Thalley, I think it is better to refactor it as I laid out in my suggestion

Reply:
@kapi-no I refactored it, and added the modifications in a new commit.